tools/mo/openvino/tools/mo/ops/parameter.py

```diff
@@ -75,6 +76,22 @@ def reverse_infer(node: Node):
         # update node 'shape' attribute (if it is not defined) from the output port shape which was calculated
         # during the reverse_infer phase
         shape = node.soft_get('shape', None)
-        if shape is None and node.out_port(0).data.get_shape() is not None:
-            node['shape'] = node.out_port(0).data.get_shape()
-
+        name = node.soft_get('name', node.id)
+        if shape is not None:
+            return
+
+        # choose dimension with the least number of dynamic dimensions
```

Comment:
I think we can deduce shape dimensions from different output ports and not only select with minimum number of dynamic dimension.

Reply:
Agreed and implemented routine `clarify_partial_shape` to do that
https://github.com/openvinotoolkit/openvino/blob/d0fd7a5aae2d0aedf7a1ba97e5877e460e219506/tools/mo/openvino/tools/mo/front/common/partial_infer/utils.py#L338